### DISCUSSION TITLE
Clickhouse ODBC Engine truncates BIGINT unsigned values exceeding 2^63

### DISCUSSION THREAD

**Discussion by alexryndin** [ClickHouse/ClickHouse#31709] [General]
How to reproduce:
1. Setup mysql server and create table:
```
CREATE TABLE `test` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `big_value` bigint(20) unsigned NOT NULL DEFAULT '0',
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8
```

2. Insert number > 2^63:
```
insert into test (big_value) value (9223372038888888888);
```
3. Setup odbc driver. I use `mysql-connector-odbc-5.2.5-8.el7.x86_64`
4. Try to retrieve data: 
``` 
SELECT *
FROM odbc('DSN=test-dsn', 'click', 'test')
LIMIT 1

Query id: ab2ca39c-3c6e-407b-9535-74fc9b60c454

┌id─┬───────────big_value─┐
│ 1 │ 9223372036854775807 │
└───┴─────────────────────┘

1 rows in set. Elapsed: 0.023 sec.
```
clickhouse version 21.8.11.4
Is it clickhouse bug or odbc issue?

**Comment by UnamedRus** [comment]
Which datatype being used for ClickHouse value?

```
SELECT *, toTypeName(big_value)
FROM odbc('DSN=test-dsn', 'click', 'test')
LIMIT 1
```

There is mysql table function also

https://clickhouse.com/docs/en/sql-reference/table-functions/mysql/

**Comment by alexryndin** [comment]
@UnamedRus none datatype is used for ClickHouse value, I just run command and get wrong data. If I create table with ODBC engine, I can explicitly set UInt64 datatype for `big_value` field, but I still get wrong data.
Native MySQL engine is not an option.

**Reply by UnamedRus** [reply]
Can you show result of my query?

**Reply by alexryndin** [reply]
😅
It's `Int64`
```
┌─id─┬───────────big_value─┬─toTypeName(big_value)─┐
│  1 │ 9223372036854775807 │ Int64                 │
└────┴─────────────────────┴───────────────────────┘

```
